avocado/utils/configure_network.py

```diff
@@ -35,168 +37,248 @@ class NWException(Exception):
     """
 
 
-def set_ip(ipaddr, netmask, interface, interface_type=None):
+class NetworkInterface:
     """
-    Gets interface name, IP, subnet mask and creates interface
-    file based on distro.
+    class For network interface
```

Comment:
We need a better docstring here.

Please, let's try to follow  PEP-0257:

https://www.python.org/dev/peps/pep-0257/